tests/unit/test_grid.py

```diff
@@ -349,24 +354,58 @@ def test_grid_width_message_with_width(test_message_with_height_width):
     assert expected_grid_width == actual_grid_width
 
 
-def test_get_geographic_area_extent():
-    """Verify extent behavior.
+def test_get_geographic_area_extent_touches_pole():
+    """Verify getting geographic extent fetches corners of grid.
 
-    extent is just defined:
-    (np.min(lons), np.min(lats), np.max(lons), np.max(lats))
+    Use a 2x3 100km grid with the lower left corner on the pole in the EASE2
+    Northern Projection
+
+    EASE2 Northern projection touches the north pole at (0., 0.)
 
     """
-    mock_area_def = Mock()
-    lats = np.array([-10, 0, 10, -20])
-    lons = np.array([-170, -175, -160, -170])
+    source_crs = CRS.from_epsg(code=6931)
+
+    #   LL, UR in meters.
+    area_extent = (0.0, 0.0, 300000.0, 200000.0)
+    area_definition = create_area_def(
+        'test area',
+        source_crs,
+        area_extent=area_extent,
+        shape=(2, 3),
+        resolution=(100000.0, 100000.0),
+    )
+    expected_geographic_extent = (0.0, 86.7714, 180.0, 90.0)
+
+    geographic_extent = get_geographic_area_extent(area_definition)
+
+    ## Pole in included is at Latitude 90.
+    assert geographic_extent[3] == 90.0
+    assert geographic_extent == approx(expected_geographic_extent, abs=1e-4)
+
+
+def test_get_geographic_area_extent_surrounds_pole():
+    """Verify EASE Grid surrounding Pole will include it.
 
-    mock_area_def.get_lonlats.return_value = (lons, lats)
+    Use whole NL Grid for simplicity
+    """
+    source_crs = CRS.from_epsg(code=6931)
 
-    expected_area_extent = (-175, -20, -160, 10)
+    #   LL, UR in meters.
+    area_extent = (-9000000.0, -9000000.0, 9000000.0, 9000000.0)
+    area_definition = create_area_def(
+        'test area',
+        source_crs,
+        area_extent=area_extent,
+        shape=(180, 180),
+        resolution=(100000.0, 100000.0),
+    )
+    expected_geographic_extent = (-179.68169, -84.63404, 180.0, 90.0)
```

Comment:
you would think this would be -180, 180, -90 to 90, but nope, this is what happens on the EASE2 grid when you go beyond the equator in projections.